tests/search/apps/test_AppResultItem.py

```diff
@@ -29,23 +39,32 @@ def app_stat_db(self, mocker):
         get_instance.return_value = mock.create_autospec(AppStatDb)
         return get_instance.return_value
 
-    def test_get_name(self, item):
-        assert item.get_name() == 'TestAppResultItem'
+    def test_get_name(self, app1):
+        assert app1.get_name() == 'TrueApp - Full Name'
+
+    def test_get_description(self, app1):
+        assert app1.get_description(Query('q')) == 'Your own yes-man'
+
+    def test_get_icon(self, app1):
+        assert isinstance(app1.get_icon(), GdkPixbuf.Pixbuf)
+        assert app1.get('Icon') == 'dialog-yes'
 
-    def test_get_description(self, item):
-        assert item.get_description(Query('q')) == 'Description of TestAppResultItem'
+    def test_search_score(self, app1):
+        assert app1.search_score("true") > app1.search_score("trivago")
 
-    def test_get_icon(self, item):
-        assert item.get_icon() == 'icon123'
+    def test_search(self, app1, app2):
+        searchresults = AppResultItem.search('false', min_score=0, apps=[app1._app_info, app2._app_info])
```

Comment:
I know this is kind of ugly (using the _app_info prop to get the appinfo object from the AppResultItem), but the only reason this `apps`parameter exists is so we can make a test like this. And I would prefer to keep the tests ugly rather than make the API harder to use, or adding methods we don't need.

Reply:
Pytest and mocking is an area of Python I'm not well acquainted with, but I'll try to do that instead :+1:

Reply:
Mocking Gio.DesktopAppInfo.get_all instead in the latest push (only the last commit is new, although the previous ones were rebased over v6).